lib/matplotlib/collections.py

```diff
@@ -1402,9 +1402,7 @@ def get_positions(self):
         '''
         segments = self.get_segments()
         pos = 0 if self.is_horizontal() else 1
-        positions = []
-        for segment in segments:
-            positions.append(segment[0, pos])
+        positions = [segment[0, pos] for segment in segments]
```

Comment:
The intermediate variables do not add any value anymore. So you can further simplify:

```suggestion
        return [segment[0, pos] for segment in self.get_segments()]
```